test/core/tsi/crl_ssl_transport_security_test.cc

```diff
@@ -296,31 +249,25 @@ class CrlSslTransportSecurityTest
       delete self;
     }
 
-    static char* LoadFile(absl::string_view file_path) {
-      grpc_slice slice;
-      GPR_ASSERT(grpc_load_file(file_path.data(), 1, &slice) ==
-                 absl::OkStatus());
-      char* data = grpc_slice_to_c_string(slice);
-      grpc_slice_unref(slice);
-      return data;
-    }
-
     static struct tsi_test_fixture_vtable kVtable;
 
     tsi_test_fixture base_;
-    bool use_revoked_server_cert_;
-    bool use_revoked_client_cert_;
-    bool use_faulty_crl_directory_;
-    bool use_revoked_intermediate_;
-    bool use_missing_intermediate_crl_;
-    bool use_missing_root_crl_;
     char* root_cert_;
     tsi_ssl_root_certs_store* root_store_;
-    tsi_ssl_pem_key_cert_pair* revoked_pem_key_cert_pairs_;
-    tsi_ssl_pem_key_cert_pair* valid_pem_key_cert_pairs_;
-    tsi_ssl_pem_key_cert_pair* revoked_intermediate_pem_key_cert_pairs_;
     tsi_ssl_server_handshaker_factory* server_handshaker_factory_;
     tsi_ssl_client_handshaker_factory* client_handshaker_factory_;
+
+    char* server_key_;
```

Comment:
Can we replace all of these `char*` and `const char*` with `std::string`'s?

Reply:
It would add a lot of messiness - the APIs these are interacting with expect char*

But I could change `LoadFile` to return a std::string, store these as std::string, then use `.c_str()` everywhere that they are assigned. WDYT?

Reply:
Do all of these APIs take ownership of the char*? If yes, I can see why it would cause messiness, but otherwise this is not clear to me.

Reply:
They all get passed to `tsi_ssl_pem_key_cert_pairs`
You can see in the destructor of the fixture in this file how we handle the destruction - we free each char* then free the allocated memory for the `tsi_ssl_pem_key_cert_pairs[]`.

So I suppose the answer is no, they don't take ownership of the char*

By messiness I mean well just have to throw `.c_str()` all over the test file

Reply:
I converted everything except `crl_directory_` to `std::string` in commit hash `7a9239a3bf`, it's important that `crl_directory_` be able to be `nullptr`. Figured it was easy to do that directly instead of having to having blocks like
```
if (crl_directory.empty()) {
 opts.crl_directory = nullptr
 else {
opts.crl_directory = crl_directory
}
```

I think this also results in more copies of the strings representing the entire files, but I don't think anything is particularly large. PTAL and let me know what you think, it'll be easy to revert if we want to keep the `char*`.

It did make the memory management cleaner, `ccbc5301e3` I removed all the handling for freeing the `char*`s